### DISCUSSION TITLE
If IW cell ends in assignment, print value

### DISCUSSION THREAD

**Discussion by janosh** [microsoft/vscode-jupyter#9533] [New Ideas]
Would be a nice feature to have the IW print variables if the last line in a cell was an assignment. Essentially replicate the behavior of

```py
# %%
(x := 5)
>>> 5
```

without needing to use the walrus operator.

In dataframe manipulation it's often helpful to verify results before proceeding. Currently this requires code like this:

```py
# %%
import pandas as pd


df = pd._testing.makeMixedDataFrame()
df

>>>  | A | B | C | D
-- | -- | -- | -- | --
0.0 | 0.0 | foo1 | 2009-01-01
1.0 | 1.0 | foo2 | 2009-01-02
2.0 | 0.0 | foo3 | 2009-01-05
3.0 | 1.0 | foo4 | 2009-01-06
4.0 | 0.0 | foo5 | 2009-01-07

# %%
df_sum = df[["A", "B"]].sum()
df_sum

A    10.0
B     2.0
dtype: float64
```

Writing just this but getting the same output would be much better.

```py
# %%
import pandas as pd


df = pd._testing.makeMixedDataFrame()

# %%
df_sum = df[["A", "B"]].sum()
```

**Comment by rchiodo** [comment]
Sounds like a cool feature idea. We'd probably have to inject code into the kernel to do so.

**Reply by janosh** [reply]
@rchiodo Btw, this would also address a small pain point I've been having with figures. If they're assigned, you currently need to call `fig.show()` explicitly to see them:


```py
import plotly.express as px

fig = px.strip(df, x="crystal_sys", y="dielectric_const")
fig.show()
```
